src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -217,6 +220,7 @@ public CompilationTracker FreezePartialStateWithTree(SolutionState solution, Doc
                 // if we already have a final compilation we are done.
                 if (inProgressCompilation != null && state is FinalState)
                 {
+                    SolutionLogger.UseExistingFullProjectState();
```

Comment:
considering this method didn't change for several years. I doubt that. also, that just change what needs to be considered like now one must know that return valued must be logged by all callers.

anyway, what it is doing here not much different than what others are doing. so fine if you want to update all those to whatever design you think better. people will review and accept the PR if team feel that's good clean up.

Reply:
> considering this method didn't change for several years

This makes it more important IMO.  Because by the time the next person comes to work on this, it will be less likely that people remember the telemetry requirements.

Reply:
> anyway, what it is doing here not much different than what others are doing. so fine if you want to update all those to whatever design you think better.

This feedback is about this particular change.  I think it would be worthwhile as there is already a lot of complexity here, and it seems more difficult than necessary to establish if the code is doing things correctly.  A simple refactoring here makes the code clearer and less difficult to get wrong.  Thanks!

>  also, that just change what needs to be considered like now one must know that return valued must be logged by all callers.

Yes.  That's the point :)  Now the language and compiler force you to think about that side of things.  That's a good thing :)  It means that if someone comes in and adds/changes any flow control, they're forced to consider the telemetry implications.

> anyway, what it is doing here not much different than what others are doing.

Lots of code follows the caller/callee pattern to ensure that the code doesn't need all control flow paths to be annotated.  Several other PRs in the last month got this exact feedback to make things clearer.  Thanks! :)